### PR TITLE
Json

### PR DESCRIPTION
**Pull request by mattstoker**
Basic JSON conversion using the JSON Parser library. Might need to move to custom conversion to avoid redundant object creation later.

### CODE REVIEW

File: Source/Core/HCCore.h [hunk 1]
```diff
@@ -60,6 +60,7 @@ void HCIntegerPrint(HCInteger self, FILE* stream);
 //----------------------------------------------------------------------------------------------------------------------------------
 // MARK: - Real Operations
 //----------------------------------------------------------------------------------------------------------------------------------
+HCBoolean HCRealIsSimilar(HCReal self, HCReal other, HCReal dissimilarity);
```
Comment (bscothern) [inline Source/Core/HCCore.h:63]: Should we renamed `dissimilarity` to `maxDissimilarity` and have use `<=`?

File: Source/Graphic/HCPrimitive.c [hunk 2]
```diff
@@ -0,0 +1,9 @@
+//
+//  HCPrimitive.c
```
Comment (bscothern) [inline Source/Graphic/HCPrimitive.c:2]: Were these files intended to be part of the `Graphics/` section? It seems like `HCPrimitive` could be a good name for some of the things in `HCCore` if we wanted to pull them out.

I also ask because there isn't much in these files.

**Review by bscothern — COMMENTED** [review]
Had some questions but I am good with all of this if you are.

File: Source/Graphic/HCPrimitive.c (hunk 2 above)
Comment (mattstoker) [reply Source/Graphic/HCPrimitive.c:2]: Yep. It has had more added as well in the next PR.

File: Source/Core/HCCore.h (hunk 1 above)
Comment (mattstoker) [reply Source/Core/HCCore.h:63]: I'll look into it.
